src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProject.cs

```diff
@@ -160,6 +160,7 @@ public async Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, stri
             switch (name)
             {
                 case "AssemblyName": _project.AssemblyName = value; break;
+                case "IntermediateAssembly": _project.CompilationOutputAssemblyFilePath = GetFullyQualifiedPath(valueOrNull); break;
```

Comment:
`GetFullyQualifiedPath` has a bug, where it won't work for a relative path, because it does `Path.Combine(_project.FilePath, value)`, but `_project.FilePath` is the full path to the actual project file (ie, `.csproj` file). When `value` is an absolute path, then that will be returned and all will be good, but the intermediate output path is a relative path, so we'd end up with `CompilationOutputAssemblyFilePath` being something like `C:\Goo\Bar.csproj\obj\Debug\net8.0` which is not correct.

Reply:
Ah I see, that was your comment about why it wasn't completely broken. The paths are coming in as absolute path so `Path.Combine` is just using the second parameter. I think what I did will fix the issue